Preprocessing/JSONProcessor.py

```diff
@@ -0,0 +1,109 @@
+import argparse
+import json
+from pathlib import Path
+import sys
+from typing import (
+    Any,
+    Dict,
+    List,
+)
+
+from Preprocessing.ErrorHandlingLogger import ErrorHandlingLogger
+from Preprocessing.utils import setup_logger
+
+
+class JSONProcessor:
+    DEFAULT_KEYS_TO_REMOVE: List[str] = ["tokens", "no_speech_prob", "compression_ratio", "avg_logprob", "temperature"]
+    UNICODE_TO_POLISH_MAP: Dict[str, str] = {
+        '\\u0105': 'ą', '\\u0107': 'ć', '\\u0119': 'ę', '\\u0142': 'ł',
+        '\\u0144': 'ń', '\\u00F3': 'ó', '\\u015B': 'ś', '\\u017A': 'ź',
+        '\\u017C': 'ż', '\\u0104': 'Ą', '\\u0106': 'Ć', '\\u0118': 'Ę',
+        '\\u0141': 'Ł', '\\u0143': 'Ń', '\\u00D3': 'Ó', '\\u015A': 'Ś',
+        '\\u0179': 'Ź', '\\u017B': 'Ż',
+    }
+
+    def __init__(self, input_folder: str, output_folder: str, extra_keys_to_remove: List[str]):
+        self.input_folder: Path = Path(input_folder)
+        self.output_folder: Path = Path(output_folder)
+        self.keys_to_remove: List[str] = self.DEFAULT_KEYS_TO_REMOVE + extra_keys_to_remove
+        self.logger: ErrorHandlingLogger = ErrorHandlingLogger(
+            class_name=self.__class__.__name__,
+            logger=setup_logger(self.__class__.__name__),
+        )
+
+    def run(self) -> int:
+        self.logger.info("Starting JSON processing...")
+        try:
+            if not (self.input_folder.exists() and self.input_folder.is_dir()):
+                self.logger.error(f"Invalid input folder path: {self.input_folder}")
+                return 1
+            self.output_folder.mkdir(parents=True, exist_ok=True)
+            self.copy_and_process_hierarchy()
+        except Exception as e: # pylint: disable=broad-exception-caught
+            self.logger.error(f"Critical error in run: {e}")
+        return self.logger.finalize()
+
+    def copy_and_process_hierarchy(self) -> None:
+        for item in self.input_folder.rglob('*'):
+            relative_path = item.relative_to(self.input_folder)
+            target_path = self.output_folder / relative_path
+
+            if item.is_dir():
+                target_path.mkdir(parents=True, exist_ok=True)
+            elif item.is_file() and item.suffix == ".json":
+                self.process_json_file(item, target_path)
+            else:
+                self.logger.info(f"Skipping unsupported file: {item}")
```

Comment:
```suggestion
                self.logger.error(f"Skipping unsupported file: {item}")
```

Opcjonalne, ale chyba nie powinno tak być XD No chyba że masz już gotową strukturę folderów która to przewiduje to resolve bez aplikowania